fastestimator/estimator/estimator.py

```diff
@@ -143,13 +148,24 @@ def _sort_traces(self):
 
     def _warmup(self):
         mode_list = self.pipeline.mode_list
+        self.total_train_steps = 0
         for mode in mode_list:
             epochs_pipeline = self.pipeline.dataset_schedule[mode].keys
             epochs_network = self.network.op_schedule[mode].keys
             signature_epochs = list(set(epochs_pipeline) | set(epochs_network))
+            if mode == "train":
+                elapse_epochs = np.diff(signature_epochs + [self.epochs])
+                assert np.all(elapse_epochs > 0), "signature epoch is not sorted correctly"
             state = {"mode": mode}
-            for epoch in signature_epochs:
+            for idx, epoch in enumerate(signature_epochs):
                 ds_iter = self.pipeline.dataset_schedule[mode].get_current_value(epoch)
+                if mode == "train":
+                    global_batch_size = self.pipeline.get_global_batch_size(epoch)
+                    if self.steps_per_epoch:
+                        max_steps = self.steps_per_epoch
+                    else:
+                        max_steps = min(self.pipeline.num_examples[mode]) // global_batch_size
```

Comment:
do we do padding on the last batch in an epoch to fill it out? if so, don't we need to round up here?

Reply:
no padding

Reply:
are we throwing out the last partial batch?

Reply:
yes.